src/circuit_breaker.erl

```diff
@@ -266,17 +284,21 @@ do_call(Service, CallFun, CallTimeout, ResetFun, ResetTimeout, Thresholds) ->
 
 -ifdef(OTP_RELEASE).
 
-preserve_exception(CallFun) ->
+preserve_exception(CallFun, Service) ->
   try
-    CallFun()
+    Start = os:timestamp(),
+    CallFun(),
+    End = os:timestamp(),
+    TimeTaken = timer:now_diff(End, Start),
+    event(ok_time_metric, Service, [{time, TimeTaken}])
   catch
     Class:Reason:Stacktrace ->
       exit({raise, Class, Reason, Stacktrace})
   end.
 
 -else.
 
-preserve_exception(CallFun) ->
+preserve_exception(CallFun, Service) ->
```

Comment:
`Service` -> `_Service` otherwise pre-OTP21 will get a compile-time warning

Reply:
Actually indicates me totally forgetting support for OTP < 21